pkg/p2p/libp2p/stream.go

```diff
@@ -29,5 +36,29 @@ func (s *stream) Headers() p2p.Headers {
 }
 
 func (s *stream) FullClose() error {
-	return helpers.FullClose(s)
+	if err := s.CloseWrite(); err != nil {
```

Comment:
While this change is extracted from previous library version, and same as before, I would posit that we actually no longer need this "hack".

As far as I could see from `github.com/libp2p/go-libp2p-core` release notes for `v0.7.0` version, previously the `Close()` function closed streams for writing, but left them open for reading. Because of this, they had "helper" function that handled fully closing the stream (to prevent leaking streams).

Since we actually do not handle anything special, and we just want to close the stream, instead of all this code, it might me enough here simply to call `Close`. In that case `Close` will not wait for any form of acknowledgment, and should free the stream object (I have not gone too deep into this).

Even with this said, I think it might be better to leave it as it was before in this PR, and have some next PR where this change will be done.

Reply:
The way we use `FullClose` actually needs the acknowledgement, since it synchronizes the flow control in certain situations, like in the handshake protocol.